player/js/utils/PropertyFactory.js

```diff
@@ -79,38 +81,63 @@ function interpolateValue(frameNum, caching) {
         keyframeMetadata.__fnct = fnc;
       }
       perc = fnc((frameNum - keyTime) / (nextKeyTime - keyTime));
-      var distanceInLine = bezierData.segmentLength * perc;
 
-      var segmentPerc;
-      var addedLength = (caching.lastFrame < frameNum && caching._lastKeyframeIndex === i) ? caching._lastAddedLength : 0;
-      j = (caching.lastFrame < frameNum && caching._lastKeyframeIndex === i) ? caching._lastPoint : 0;
-      flag = true;
-      jLen = bezierData.points.length;
-      while (flag) {
-        addedLength += bezierData.points[j].partialLength;
-        if (distanceInLine === 0 || perc === 0 || j === bezierData.points.length - 1) {
-          kLen = bezierData.points[j].point.length;
-          for (k = 0; k < kLen; k += 1) {
-            newValue[k] = bezierData.points[j].point[k];
+      const startPoint = keyData.s;
+      const endPoint = nextKeyData.s || keyData.e;
+
+      if (perc > 1) {
+        let tangent = keyData.ti;
+        if (Math.hypot(...tangent) <= Number.EPSILON) {
```

Comment:
nit: Might make sense to compute `Math.hypot(...tangent)` once and reuse it below, to avoid doing 2 `sqrt`s when one would suffice?